### PR TITLE
[1LP][WIPTEST] Move cfme.Credential to cfme.base.credentials

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
part 1 of emptying out the cmfe package namespace from imports

split pull request because the larger one keeps having failures and mistakes

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 495c6fcd14df1ac254324611748f5321bb79db29:

`cfme/base/credentials.py`:
- :heavy_exclamation_mark: Line 33:55: **W292** *no newline at end of file*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/infrastructure/config_management.py [hunk 1]
```diff
@@ -250,7 +251,7 @@ def load_from_yaml(cls, key):
             name=data['name'],
             url=data['url'],
             ssl=data['ssl'],
-            credentials=cls.Credential(
+            credentials=cfme.base.credentials.Credential(
```
Comment (psav) [inline cfme/infrastructure/config_management.py]: **REQUIRED:** This should still be ```cls.``` I think as it uses the providers ```Credential``` object, not the base one

File: cfme/infrastructure/config_management.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 
 import cfme
+import cfme.base.credentials
```
Comment (psav) [inline cfme/infrastructure/config_management.py]: **REQUIRED:** As this is a new line and in ALL the other files we have imported the object and not the module, can we do this here too perhaps using an ```as``` to distinguish?

File: cfme/common/provider.py [hunk 3]
```diff
@@ -3,6 +3,7 @@
 from manageiq_client.api import APIException
 
 import cfme
+import cfme.base.credentials
```
Comment (mshriver) [inline cfme/common/provider.py]: Why import module here instead of just the Credential class? (Looks like the class is just used for inheritence here.

Comment (izapolsk) [inline cfme/common/provider.py]: OPTIONAL: @RonnyPfannschmidt please follow the way you used below - just import class as BaseCredential

File: cfme/infrastructure/host.py [hunk 4]
```diff
@@ -161,7 +162,7 @@ def _form_mapping(self, create=None, **kwargs):
                 'ipmi_address_text': kwargs.get('ipmi_address'),
                 'mac_address_text': kwargs.get('mac_address')}
 
-    class Credential(cfme.Credential, Updateable):
+    class Credential(cfme.base.credentials.Credential, Updateable):
```
Comment (izapolsk) [inline cfme/infrastructure/host.py]: ^^

**Comment by dajoRH** [comment]
Lint report for commit 2b985c89e9f4f3074d92e5b1347e625f772517db:

`cfme/common/provider.py`:
- :red_circle: Line 5:1: **F401** *'cfme' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by psav** [comment]
Lint.....LGTM

**Pull request closed without merging**